design-proposals/host-ip-pool.md

```diff
@@ -0,0 +1,442 @@
+# Host IP pooling with NNCPs.
+
+## Summary
+
+Provide a mechanism to assign static IP configuration for secondary interfaces
+for cluster nodes, supporting cluster scaling up and down.
+
+## Motivation
+
+Using DHCP as the IP assignment mechanism has some drawbacks:
+- Introduce latencies on node restart.
+- It presents a single point of failure in case the DHCP server is failing
+- It depends on network setup that can be complex.
+
+To overcome the above-mentioned drawbacks, statically configured interfaces taking the address from a
+dynamic host IP pool is a solution. Since the interfaces are static there is no
+reconfiguration after some time or at node restart.
+
+This functionality was requested on U/S via https://github.com/nmstate/kubernetes-nmstate/issues/725
+
+Although this proposal will present a solution with an agnostic storage
+mechanism, following is a spike done using whereabouts at the storage
+https://github.com/nmstate/kubernetes-nmstate/pull/1081
+
+
+### Similar product
+
+- Metal3 IPAM: https://metal3.io/blog/2020/07/06/IP_address_manager.html
+  - It depends on CAPI (ClusterAPI)
+  - It's though for Metal3 baremetal.
+- Azure private addresses: https://docs.microsoft.com/en-us/azure/virtual-network/ip-services/private-ip-addresses
+  - Specific to azure
+  - Some code has to be put in place to consume it.
```

Comment:
Can you add a reference to how assisted installer does this on day 1 ?

Reply:
I'm told it's somewhere in https://github.com/openshift/assisted-service/blob/9bae4c3662e4755e80251284aba529318d76c8e9/docs/user-guide/restful-api-guide.md#additional-nmstate-configuration-examples